src/modules/Remote.rb

```diff
@@ -135,18 +152,30 @@ def Read
         from: "any",
         to:   "list <map>"
       )
-      vnc_conf = Builtins.filter(xinetd_conf) do |m|
-        s = Ops.get_string(m, "service", "")
-        s == "vnc1" || s == "vnchttpd1"
+
+      xinetd_vnc1_enabled = false
+      xinetd_vnchttp1_enabled = false
+      xinetd_conf.each do |m|
+        xinetd_vnc1_enabled = m["enabled"] if m["service"] == "vnc1"
+        xinetd_vnchttp1_enabled = m["enabled"] if m["service"] == "vnchttpd1"
       end
-      vnc = Builtins.size(vnc_conf) == 2 &&
-        Ops.get_boolean(vnc_conf, [0, "enabled"], false) &&
-        Ops.get_boolean(vnc_conf, [1, "enabled"], false)
 
       log.info "#{XDM_SERVICE_NAME}: #{xdm}, DM_R_A: #{dm_ra}"
-      log.info "xinetd: #{xinetd}, VNC: #{vnc}"
+      log.info "xinetd: #{xinetd}, VNC1: #{xinetd_vnc1_enabled}, VNCHTTP1: #{xinetd_vnchttp1_enabled}"
+
+      vncmanager = Service.Enabled(VNCMANAGER_SERVICE)
 
-      @allow_administration = xdm && dm_ra && xinetd && vnc
+      if xdm && dm_ra && xinetd && xinetd_vnchttp1_enabled
```

Comment:
It does behave the way you are describing. Which is the same way as it used to.

It we omit the `vnchttpd1` from this condition, it will get re-enabled after submitting the form.

I don't have any preferences either way, if you think it's better to omit it here, I will change it.